tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -121,28 +162,90 @@ private static void AnnotateSpan(Span span)
             }
         }
 
+        private static void LogMatchesIfDebugEnabled(WafMatch[] results, bool blocked)
+        {
+            if (Log.IsEnabled(LogEventLevel.Debug))
+            {
+                for (var i = 0; i < results.Length; i++)
+                {
+                    var result = results[i];
+                    if (blocked)
+                    {
+                        Log.Debug("Blocking current transaction (rule: {RuleId})", result.Rule);
+                    }
+                    else
+                    {
+                        Log.Debug("Detecting an attack from rule {RuleId}", result.Rule);
+                    }
+                }
+            }
+        }
+
+        private static void AddHeaderTags(Span span, IHeadersCollection headers, Dictionary<string, string> headersToCollect)
+        {
+            var tags = SpanContextPropagator.Instance.ExtractHeaderTags(headers, headersToCollect, defaultTagPrefix: SpanContextPropagator.HttpResponseHeadersTagPrefix);
+            foreach (var tag in tags)
+            {
+                span.SetTag(tag.Key, tag.Value);
+            }
+        }
+
+        private static Span GetLocalRootSpan(Span span)
+        {
+            var localRootSpan = span.Context.TraceContext?.RootSpan;
+            return (localRootSpan == null) ? span : localRootSpan;
+        }
+
+        private static void AddEndPoint(Span span)
```

Comment:
[nit] maybe TryAddEndpoint as the tag may not be set.

Reply:
Fixed in 2403b6fd